### DISCUSSION TITLE
Unable to pre-populate multi select picker from data

### DISCUSSION THREAD

**Discussion by jealouscloud** [Budibase/budibase#9404] [Help] [answered]
I have a multi select and have tried auto-populating its default value with javascript similar to:

return $("draft_groups_data.Rows").map(x=>x.value).join(",")

which outputs:

```
group1,group2
```
The budibase devtools report accurately that the default value is "group1,group2".

In spite of their name/value equal to that of options in the multi-select, it does not get reflected by the picker.

If I manually return the string value with no logic:

```
return "group1, group2"
```

it works. The same is true if I manually enter the value. 

What's going on here? Are there successful examples of pre populating a multi-select?

**Comment by melohagan** [comment]
Hey @jealouscloud

Could you share some screenshots of your component tree and bindings please? Just to make sure I'm on the same page.

**Comment by jealouscloud** [comment, marked as answer]
I was able to resolve this by using ID as the 'value' option in the picker.

I had to make my map return id.toString() instead of the int. Then it started working as long as I hid the component before I set the state variable I use in the default value field.

**Reply by peterhuppertz** [reply]
> Then it started working as long as I hid the component before I set the state variable I use in the default value field.

Thanks for that. I had the same problem retrieving the value from the DB. Once I set the Condition "Hide if empty" to the picker it pre-populated correctly and showed up. Probably a bug?!

**Comment by oanylund** [comment]
Is this a bug? Experiencing the same. Works if I return a static string array, but not when using dynamic data.
Quite sure i had this working  in an earlier version, but not now with version: v2.3.16
